File: i2f-jdk/i2f-swl/src/main/java/i2f/swl/data/SwlContext.java
package i2f.swl.data;

import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author Ice2Faith
 * @date 2024/7/10 14:11
 * @desc
 */
@Data
@NoArgsConstructor
public class SwlContext {
    private String remotePublicKey;
    private String remoteAsymSign;

    private String selfPrivateKey;
    private String selfAsymSign;

    private String timestamp;
    private String nonce;
    private String key;
    private String randomKey;
    private String data;
    private String sign;
    private String digital;

    private String clientId;
    private String currentTimestamp;
    private String window;
    private String nonceKey;
    private boolean signOk;
    private boolean digitalOk;
}

File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-turbo-web/i2f-swl/core/data/SwlContext.js
import AsymKeyPair from "../../../i2f-core/crypto/asymmetric/AsymKeyPair";

/**
 * @reutrn {SwlContext}
 * @constructor
 */
function SwlContext() {
    /**
     * @type {String}
     */
    this.remotePublicKey = null
    /**
     * @type {AsymKeyPair}
     */
    this.remoteAsymSign = null
    /**
     * @type {String}
     */
    this.selfPrivateKey = null
    /**
     * @type {String}
     */
    this.selfAsymSign = null
    /**
     * @type {String}
     */
    this.timestamp = null
    /**
     * @type {String}
     */
    this.nonce = null
    /**
     * @type {String}
     */
    this.key = null
    /**
     * @type {String}
     */
    this.randomKey = null
    /**
     * @type {String}
     */
    this.data = null
    /**
     * @type {String}
     */
    this.sign = null
    /**
     * @type {String}
     */
    this.digital = null
    /**
     * @type {String}
     */
    this.clientId = null
    /**
     * @type {String}
     */
    this.currentTimestamp = null
    /**
     * @type {String}
     */
    this.window = null
    /**
     * @type {String}
     */
    this.nonceKey = null
    /**
     * @type {boolean}
     */
    this.signOk = null
    /**
     * @type {boolean}
     */
    this.digitalOk = null
}

export default SwlContext
